akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -629,6 +634,23 @@ private[akka] object Running {
         Tagged(adaptedEvent, tags)
     }
 
+    // note that this shadows tryUnstashOne in StashManagement from HandlingCommands
+    private def tryUnstashOne(behavior: Behavior[InternalProtocol]): Behavior[InternalProtocol] = {
+      if (isStashEmpty) {
+        recursiveUnstashOne = 0
```

Comment:
The failing tests captured an oversight. The recursion counter wasn't reset in enough places. Now I'm maybe resetting it in more places than actually needed, but there are multiple paths to tryUnstashOne, including onPublishedEvent and onReplicatedEvent. Not very happy with the solution but it's rather involved to begin with.

It would be nice with a ThreadLocal counter but that has some overhead and not a tool we normally use.